apps/web/src/components/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+import { useRef } from 'react';
+import './SearchBar.scss';
+import SearchIcon from '../../assets/searchIcon.svg';
+
+export interface Props {
+  /** What happens when we use a searchstring to search for results? */
+  handleSearch: (searchString: string) => void;
+  /** What happens when a user begins typing what they're searching for? */
+  filterResults: (searchString: string) => object[];
+  /** An optional custom test ID for the component.  */
+  customTestId?: string;
+}
+
+/**
+ * A component that renders a SearchBar
+ *
+ * @param {Object} props - The component props
+ * @param {function} handleSearch - Invoked when the user presses enter
+ * @param {function} filterResults - Invoked when the user begins typing into input
+ * @param {string} customTestId - An optional custom test ID for the component
+ * @returns {React.ReactElement} - The rendered component
+ */
+const SearchBar = ({
+  handleSearch,
+  filterResults,
+  customTestId = 'searchbar',
+}: Props): React.ReactElement => {
+  /** references input element */
+  const ref = useRef<HTMLInputElement>(null);
+
+  /**
+   * Handles a keydown event on the searchbar input
+   *
+   * @param {React.KeyboardEvent} - Keydown event.
+   * @returns {void}
+   */
+  const handleKeyDown = (
+    event: React.KeyboardEvent<HTMLInputElement>
+  ): void => {
+    if (event.key === 'Enter') {
+      const searchString = ref.current?.value;
+      if (
+        searchString !== undefined &&
+        searchString !== null &&
+        searchString !== ''
```

Comment:
Since you use these sets of conditions 3 times would it make sense to save them in a variable for reuse? I think it's fine and maybe clearer the way as it is though.

Reply:
This is like the only time I recommend using the `!!` operator: 

<img width="446" alt="Screen Shot 2023-04-02 at 3 12 13 PM" src="https://user-images.githubusercontent.com/12022922/229373872-59423092-eca0-4bbb-ad84-afdeab4773e2.png">

Reply:
So this  could all be simplified to `!!searchString`. If you want something more semantic you could return a boolean from a function like:

```ts
const isSearchStringValud = (searchString : string) : boolean => return !!searchString;
```